src/main/daml/Daml/Finance/Claims/Lifecycle/Rule.daml

```diff
@@ -97,38 +88,26 @@ template Rule
           electionTime = Election.getElectionTime election
           election = electionEvent electionTime v.electorIsOwner v.claim
 
-        -- retrieve election data for previous elections and re-create election events
-        nv <- BaseInstrument.exerciseInterfaceByKey @DynamicInstrument.I v.instrument lifecycler
-          DynamicInstrument.GetView with viewer = lifecycler
-        let
-          prevElections = map (\e ->
-            case e.election of
-              None -> error "election missing"
-              Some (electorIsOwner, tag) -> electionEvent e.t electorIsOwner tag)
-            nv.prevElections
-
-        -- fetch claim tree as of instrument inception
-        claimInstrument <- fetchInterfaceByKey @BaseInstrument.R v.instrument
-        let acquisitionTime = Claim.getAcquisitionTime claimInstrument
-
-        -- fast-forward the claims tree to the current version by replaying the previous elections
-        claims <- fst <$>
-          lifecycle lifecycler observableCids claimInstrument prevElections
+        (claimsWithoutElection, pending, claims, claimInstrument, nv) <-
+          fastForwardAndLifecycle v.instrument observableCids electionTime lifecycler
 
         -- lifecycle as of the time given by the event, by applying the corresponding election
+        let acquisitionTime = Claim.getAcquisitionTime claimInstrument
         (remaining, pending) <- lifecycleClaims observableCids acquisitionTime claims [election]
+        assertMsg "Election not possible on this date" $ remaining /= claimsWithoutElection
```

Comment:
I would add a comment summarizing why this check makes sense to ensure that no election has taken place